File: 1127-last-stone-weight/1127-last-stone-weight.js
/**
 * @param {number[]} stones
 * @return {number}
 */
var lastStoneWeight = function(stones) {
    while (stones.length > 1) {
        stones.sort((a, b) => a - b);
        if (stones[stones.length - 1] === stones[stones.length - 2]) {
            stones.pop();
            stones.pop();
        } else {
            stones[stones.length - 2] = stones[stones.length - 1] - stones[stones.length - 2];
            stones.pop();
        }
    }
    if (stones.length === 1) {
        return stones[0];
    } if (stones.length === 0) {
        return 0
    }
    
};

File: 1127-last-stone-weight/1127-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        neg = [-x for x in stones]
        heapq.heapify(neg)
        while len(neg) > 1:
            s1 = heapq.heappop(neg)
            s2 = heapq.heappop(neg)
            s1 = s1 - s2
            print(s1)
            if s1 < 0:
                heapq.heappush(neg, s1)
                
        if len(neg) == 1:
            return -neg[0]
        else:
            return 0